### DISCUSSION TITLE
How do I render a page?

### DISCUSSION THREAD

**Discussion by Ferks-FK** [GrapesJS/grapesjs#6158] [Q&A]
I'm trying to retrieve the “built” page to display, but I don't seem to have this information in the DB.

```
storageManager: {
    type: 'remote',
    stepsBeforeSave: 1,
    autosave: true,
    autoload: true,
    options: {
        remote: {
            urlLoad: 'http://page-builder.localhost/api/load-page',
            urlStore: 'http://page-builder.localhost/api/save-page',
            contentTypeJson: true,
            onStore: result => {
                console.log(result)
                return ({ data: result })
            },
            onLoad: result => {
                console.log(result)
                return result.data;
            },
        }
    },
},
```

In the `onStore` function I get this:

```
{
  assets: [{...}],
  pages: [{...}],
  styles: [{...}],
  symbols: []
}
```

And that's what's being stored in the DB, for editing the page using Graper this works, but I can't find anything useful in this answer for rendering the page itself.

Note: I found [this](https://stackoverflow.com/a/64928663) answer, it says that Graper delivers the `gjs-html` and `gjs-css` attributes, which are not present in my case.

**Comment by mosh-tudor** [comment]
@Ferks-FK

Your HTML code is in `pages: [{...}]`, and CSS styles are in `styles: [{...}]`, so you need to find a way to display this content on your page. Everything depends on your stack.

In the end, it should look something like this:
```html
<html ...>
  <head>
   <style>
     = styles
   </style>
  </head>
  <body>
    = html
  </body>
</html>
```

**Reply by Ferks-FK** [reply]
> @Ferks-FK
> 
> Your HTML code is in `pages: [{...}]`, and CSS styles are in `styles: [{...}]`, so you need to find a way to display this content on your page. Everything depends on your stack.
> 
> In the end, it should look something like this:
> 
> ```
> <html ...>
>   <head>
>    <style>
>      = styles
>    </style>
>   </head>
>   <body>
>     = html
>   </body>
> </html>
> ```

So those properties (which I mentioned in the stackoverflow link) no longer exist?
Do I literally have to go through that data and build the page?

**Reply by mosh-tudor** [reply]
I don't think that comment is still relevant; it was written 4 years ago, when there were no pages in GrapesJS.

You have all the information in the database, and all that's left is to display it on your page.

**Comment by artf** [comment]
Here you can see an example of how to add HTML/CSS to the project JSON itself in the storage: https://grapesjs.com/docs/modules/Storage.html#common-use-cases

I'd suggest taking it only as an example of how to export the code from pages. Keep the project file separate from the HTML/CSS data, and use the exported code only in the "export"/"publish" action in your application.
